packages/style-engine/class-wp-style-engine.php

```diff
@@ -58,6 +58,9 @@ final class WP_Style_Engine {
 					'default' => 'background-color',
 				),
 				'path'          => array( 'color', 'background' ),
+				'css_vars'      => array(
```

Comment:
Just double-checked and refreshed my memory: the addition here and in `gradient` looks safe to me and doesn't affect calls from `colors.php` since in that file, we're calling `gutenberg_style_engine_get_styles` with the `convert_vars_to_classnames` option flag, which means that it skips adding the `css_vars` in cases where we intend to grab classnames instead. For example, if we add a Post Title block to a post or page and set background / gradient colors, then with this PR applied, it'll still receive classnames and not any inline styles, which is as we'd expect 👍

E.g. this is as on trunk:

<img width="1266" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/6a3e4678-6530-45c3-9e0f-4298c1d34161">

While elsewhere on the same page, the container-based element styles are applied correctly via the `background-color` property, etc:

<img width="1274" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/9948eb98-2a64-4a98-a9fb-f6de4c5d644d">

Reply:
The style engine™️ delivers